### ISSUE TITLE
Any operation is super slow

### ISSUE THREAD

**Issue by jtsom** [microsoft/azuredatastudio#25340]
Type: <b>Bug</b>

Trying to do any operation, to any database, either local, or through a VPN to our Azure databases, is super slow.

It is taking 30 - 60 seconds just to open the database server in the server explorer, or executing any statement. 

Opening a connection to a database server also takes 30 seconds to multiple minutes.

Something happened in a recent update to cause this SUPER slow down.

Azure Data Studio version: azuredatastudio-insiders 1.48.0-insider (8a715ed3b1e87153e46651b920367170564eb376, 2024-02-15T12:58:00.069Z)
OS version: Windows_NT x64 10.0.22631
Restricted Mode: No
Preview Features: Enabled
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8500 CPU @ 3.00GHz (6 x 3000)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.78GB (16.34GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (6)</summary>

Extension|Author (truncated)|Version
---|---|---
dacpac|Mic|1.14.0
import|Mic|1.6.0
schema-compare|Mic|1.21.0
ADS-add-columns|sof|0.0.12
vscodeintellicode|Vis|1.2.1
poor-sql-formatter|WSR|0.1.0

</details>

**Comment by cheenamalhotra** [comment]
Can you please share logs we can investigate when you notice a slowdown?

**Comment by jtsom** [comment]
Here you go.

[console.log](https://github.com/microsoft/azuredatastudio/files/14345290/console.log)
[20240220T084459.zip](https://github.com/microsoft/azuredatastudio/files/14345295/20240220T084459.zip)

**Comment by cheenamalhotra** [comment]
I believe you accidentally closed the issue.. reopening it. 
Also the logs don't contain any MSSQL logs, can you recheck you've enabled them as per above instructions and restarted ADS?

**Comment by jtsom** [comment]
Trying again.
[20240220T112038.zip](https://github.com/microsoft/azuredatastudio/files/14348417/20240220T112038.zip)
[console.log](https://github.com/microsoft/azuredatastudio/files/14348420/console.log)

**Comment by cheenamalhotra** [comment]
Thanks! I've gathered the logs, removed from comment above to remove server info.

**Comment by cheenamalhotra** [comment]
From the logs, I can notice a delay of 20+ seconds on every connection during login in the SqlClient library, e.g.

```
24-02-20 11:20:55.9181035 pid:19624 tid:18 sqltools Verbose: 0 : eventTID:12176 TdsParser.Connect | SEC | Connection Object Id 4, Authentication Mode: SqlPassword
24-02-20 11:21:17.0460890 pid:19624 tid:18 sqltools Verbose: 0 : eventTID:12176 <sc.TdsParser.Connect|SEC> Sending prelogin handshake
```

We will investigate and get back to you.

**Comment by jtsom** [comment]
Yep that's pretty much what happens. Happening lately on connections to my local sql server, and more severely when trying to connect to our Azure Sql database, through our VPN (which is what I did in this test.)

Thanks.

**Comment by cheenamalhotra** [comment]
Could you try to resolve server IP address (using `nslookup {hostname}` and connect to IP address directly? This will help identify if this is related to DNS service.

**Comment by jtsom** [comment]
Stil having MASSIVE slowness when going through an Azure VPN:

When connecting through an Azure VPN connection, to an Azure hosted database, ADS takes forever to connect - if it connects at all. Trying to twist open a database connection in the Connections pane most times takes several minutes (10+?) or never opens up at all.

Using SSMS, I am able to connect right up to the same database, almost instantly.

I tried getting the IP address, but nslookup reports it cannot find the server - even though SSMS, using the same hostname, works just fine.
